### PR TITLE
Update CI URLs to point to new caching infrastructure

### PR DESCRIPTION
**Pull request by staticfloat**
Hello there!
This is an automated pull request submitted by `@staticfloat` to help package authors transition their Julia installation CI setups to the new binary provider URL.  Please take a look at this PR, and if there is a problem or it doesn't download correctly when CI runs, feel free to ping `@staticfloat`

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 environment:
   matrix:
-#  - JULIAVERSION: "julialang/bin/winnt/x86/0.4/julia-0.4-latest-win32.exe"
-#  - JULIAVERSION: "julialang/bin/winnt/x64/0.4/julia-0.4-latest-win64.exe"
-  - JULIAVERSION: "julianightlies/bin/winnt/x86/julia-latest-win32.exe"
-  - JULIAVERSION: "julianightlies/bin/winnt/x64/julia-latest-win64.exe"
+  - JULIA_URL: "https://julialang-s3.julialang.org/bin/winnt/x86/0.4/julia-0.4-latest-win32.exe"
+  - JULIA_URL: "https://julialang-s3.julialang.org/bin/winnt/x64/0.4/julia-0.4-latest-win64.exe"
+  - JULIA_URL: "https://julialangnightlies-s3.julialang.org/bin/winnt/x86/julia-latest-win32.exe"
```
Comment (andyferris) [inline appveyor.yml]: It seems these haven't been updated since 0.5-dev...

Comment (staticfloat) [inline appveyor.yml]: @andyferris what do you mean?  The nightlies were updated less than two hours ago.

Comment (andyferris) [inline appveyor.yml]: No I mean my `appveyor.yml` file hasn't been updated for ages. I'll have to make some additional changes, I guess.

Comment (tkelman) [inline appveyor.yml]: doesn't look like it's enabled. the package also doesn't support 0.4 any more

**Comment by c42f** [comment]
I think this is well out of date by now. Let's close this and possibly just migrate to github actions.

**Pull request closed without merging**